### PR TITLE
lib: allow booleans in arrays for match conditions

### PR DESCRIPTION
**Pull request by MylesBorins**
This is a decent pattern to use when we need to temporarily skip a
module but don't want to lose the existing skip data for specific
platforms

### CODE REVIEW

**Comment by MylesBorins** [comment]
previously invalid input would have been ignore, updating tests and marking this semver-major

**Comment by MylesBorins** [comment]
CI failures make no sense and look like they might be infra related

File: lib/match-conditions.js [hunk 1]
```diff
@@ -54,6 +54,7 @@ function isStringMatch(conditions) {
 
 function isArrayMatch(arr) {
   return _.some(arr, (item) => {
+    if (typeof item === 'boolean' && item) return true;
```
Comment (BridgeAR) [inline lib/match-conditions.js]:
```suggestion
    if (typeof item === 'boolean') return item;
```

**Comment by richardlau** [comment]
> 
> 
> CI failures make no sense and look like they might be infra related

They were due to a bug in `esm`, which the `tap` plugin recently started using.

**Comment by richardlau** [comment]
Rerunning the Travis CI: https://travis-ci.com/nodejs/citgm/builds/99332371

**Pull request merged**